### PR TITLE
Regression: Prevent connect to asterisk when VoIP is disabled

### PR DESCRIPTION
**Pull request by murtaza98**

### CODE REVIEW

File: server/services/startup.ts [hunk 1]
```diff
@@ -29,11 +30,23 @@ api.registerService(new MeteorService());
 api.registerService(new NPSService(db));
 api.registerService(new RoomService(db));
 api.registerService(new SAUMonitorService());
-api.registerService(new VoipService(db));
-api.registerService(new OmnichannelVoipService(db));
 api.registerService(new TeamService(db));
 api.registerService(new UiKitCoreApp());
 
+let voipServiceInstance: VoipService;
```
Comment (KevLehman) [inline server/services/startup.ts:36]: Can't this be handled at service level? Conditionally registering services based on settings is something i'm not in love with

Comment (murtaza98) [reply server/services/startup.ts]: For sure, moved the logic to service level

File: server/services/voip/service.ts [hunk 2]
```diff
@@ -31,11 +32,25 @@ export class VoipService extends ServiceClassInternal implements IVoipService {
...
-		}
+		this.init();
+	}
+
+	private async init(): Promise<void> {
+		settings.watch('VoIP_Enabled', (value) => {
+			try {
+				if (value) {
+					this.logger.info('Starting VoIP service');
+					Promise.await(this.commandHandler.initConnection(CommandType.AMI));
```
Comment (KevLehman) [inline server/services/voip/service.ts:43]: You're in an async function, no need for `Promise.await` here 👀

Comment (murtaza98) [inline server/services/voip/service.ts:43]:
```suggestion
					await this.commandHandler.initConnection(CommandType.AMI);
```

Comment (murtaza98) [reply server/services/voip/service.ts]: Done :)

File: server/services/voip/service.ts [hunk 3]
```diff
@@ -31,11 +32,25 @@ export class VoipService extends ServiceClassInternal implements IVoipService {
...
-		}
+		this.init();
+	}
+
+	private async init(): Promise<void> {
+		settings.watch('VoIP_Enabled', (value) => {
+			try {
+				if (value) {
+					this.logger.info('Starting VoIP service');
+					await this.commandHandler.initConnection(CommandType.AMI);
```
Comment (KevLehman) [inline server/services/voip/service.ts:43]:
```suggestion
					Promise.await(this.commandHandler.initConnection(CommandType.AMI));
```

Comment (KevLehman) [reply server/services/voip/service.ts:43]: :(

**Comment by murtaza98** [comment]
@sampaiodiego adding you as a reviewer over here since I wanted to confirm if it is safe to call `settings.watch` within a service. Please advice :pray: Thanks in advance

**Review by sampaiodiego — APPROVED** [review]

**Pull request merged**